app/src/main/java/com/internship/move/ui/authentication/AuthenticationViewModel.kt

```diff
@@ -1,23 +1,76 @@
 package com.internship.move.ui.authentication
 
+import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
-import com.internship.move.repository.Repository
+import com.internship.move.network.model.AddLicenseResponse
+import com.internship.move.network.model.ErrorResponse
+import com.internship.move.network.model.LoginResponse
+import com.internship.move.network.model.RegisterResponse
+import com.internship.move.repository.UserRepository
+import com.internship.move.utils.Constants.LOADING_DELAY
+import com.internship.move.utils.InternalStorageManager
+import com.squareup.moshi.Moshi
+import kotlinx.coroutines.delay
+import org.koin.java.KoinJavaComponent.inject
+import retrofit2.HttpException
 
 class AuthenticationViewModel(
-    private val repo: Repository
+    private val repo: UserRepository,
+    private val internalStorageManager: InternalStorageManager
 ) : ViewModel() {
 
-    fun register(email: String, username: String, password: String) {
-        //api call
-        repo.setIsLoggedIn(true)
-    }
+    private val moshi: Moshi by inject(Moshi::class.java)
```

Comment:
I assume it works the same way, but myself I prefer using `inject<Moshi>()` instead, since it's shorter.
Also, maybe you could inject it in the constructor directly 🤔